client/agent-opentracing/src/main/resources/apmrules/org/apache/apm-apache-http-client.btm

```diff
@@ -35,3 +35,14 @@ IF hasSpan()
 DO
   finishSpan();
 ENDRULE
+
+RULE org.apache.http.client(4) Apache HttpClient Producer Sync Exception
+CLASS ^org.apache.http.impl.client.CloseableHttpClient
+METHOD doExecute
+HELPER org.hawkular.apm.agent.opentracing.OpenTracingManager
+AT EXCEPTION EXIT
+IF hasSpan()
+DO
+  getSpan().setTag("error","true").setTag("http.status_code",String.valueOf(503));
```

Comment:
Originally I just had the error tag, but thought as this is using HTTP that maybe it would be good to consistently provide a http status code. There isn't a value representing Connection Refused, but this was the closest - Service Unavailable.

Once our OT provider supports logging then I think we should have a jira to add at all relevant places  - so agree that the details could be added as log event.

Wasn't 100% sure about the status code, hence marked 'do not merge' until we had discussed it - so leave it in or take it out?